python/xorbits/deploy/kubernetes/client.py

```diff
@@ -143,6 +145,18 @@ def __init__(
         self._cluster_type = self._get_cluster_type(cluster_type)
         self._ingress_name = "xorbits-ingress"
         self._use_local_image = kwargs.pop("use_local_image", False)
+        self._external_storage = external_storage
+        if self._external_storage and self._external_storage not in ["juicefs"]:
+            raise ValueError(
+                "Currently, only juicefs is supported as one of our storage backend."
+            )
+        if self._external_storage == "juicefs":
+            self._metadata_url = kwargs.pop("metadata_url", None)
```

Comment:
As it is only for external storage, all parameters like `metadata_url` can pass to a  parameter called `external_storage_config`.